File: 1635-number-of-good-pairs/number-of-good-pairs.py
class Solution:
    def numIdenticalPairs(self, nums: List[int]) -> int:
        count = {}
        for num in nums:
            count[num] = count.get(num, 0) + 1
        
        print(count)

        p = 0

        for key in count:
            p = p + (count[key]*(count[key]-1)) // 2

        return p
        

File: 1635-number-of-good-pairs/number-of-good-pairs.java
import java.util.HashMap;
class Solution {
    public int numIdenticalPairs(int[] nums) {
        HashMap<Integer, Integer> hash = new HashMap<>();

        for(int i: nums){
            hash.put(i, hash.getOrDefault(i, 0) + 1);
        } 

        int count = 0;

        for(int i: hash.keySet()){
            int frequency = hash.get(i);
            count += (frequency * (frequency - 1)) / 2;
        }
        return count;

    }
}